week2/index.js

```diff
@@ -1,65 +1,150 @@
+        break;
+    //when the user enters the add command a new todo is added to his todoList
+    //this is done via the addTodoItem function
+    case 'add':
+        addTodoItem(todosFile,options);
+        break;
+    //if the user enter the command remove, followed by the item number, the item will be removed.
+    //the functions removeTodoItem will remove the item with the number the user entered
+    case 'remove':
+        removeTodoItem(todosFile,options);
+        break;
+    //when the user asks for list, he should get the list of todos he has
+    //the list should be numbered with human numbers
+    //the function listTodos is invoked in this case which will log the proper response
+    case 'list':
+        listTodos(todosFile);
+        break;
+    //the reset command will remove all the items in the todoList
+    //when this command is received the function resetTdosList is invoked, which will remove all the items in the todos
+    case 'reset':
+        resetTodosList(todosFile);
+        break;
+    //the update command will update the content of the task. Old contents will be removed
+    //When this command is detected the function updateTodosList is invoked
+    case 'update':
+        updateTodosList(todosFile,options);
+        break;
+    //if the user doesn't enter a valid command, he will be prombted and
+    //he will get the help file to learn how to use the commands proberly
+    default :
+        console.log('Please enter a valid command');
+        showHelp();
 }
 
-function splitStringByNewline(string) {
-  return string.split('\n').filter(function(element) {
-    element = element.trim();
-    return element.length > 0;
-  });
+//this function will read the help file and display it to the user.
+//it doesn't take any parameters but depends on the file help.txt
+function showHelp() {	
+ const helpText =  fs.readFileSync('./help.txt',{encoding:'UTF-8'});
+    console.log(helpText);
 }
 
-function showHelp() {
-  openFile('help.txt', function(error, data) {
-    if (error) {
-      return console.log('Error: the help file could not be displayed', error);
+//add an item to the todo list
+//this functions takes the todoFile json and the commands array as parameters. 
+//It adds the new items to it and at the end it will write it the new object to the Todo file
+//this way we can save our new todos and keep our old ones
+function addTodoItem(jsonObject,commands) {
+    //assign the task text to a variable.
+    let itemToAdd = commands[1];
+    //check if the user has entered text or not
+    if(itemToAdd === undefined || itemToAdd.length < 1){
+        console.log('Please enter a name for your task');
+    }else{
+        //push the text the user entered to our data file
+        jsonObject.tasks.push(itemToAdd);
+        //write the modified object to the todo File
+        console.log('Added new item');
+        jsonObject = JSON.stringify(jsonObject);
+        return fs.writeFile('todo.json',jsonObject);
```

Comment:
This results in:
`(node:5301) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.`

Why does this happen and how can we fix it?